xbmc/addons/kodi-addon-dev-kit/include/kodi/Filesystem.h

```diff
@@ -1266,6 +1266,44 @@ namespace vfs
     //==========================================================================
     ///
     /// @ingroup cpp_kodi_vfs_CFile
+    /// @brief retrieve a http response header
+    ///
+    /// @param[in] name         The name of header to retrieve the value for
+    /// @return                 value of requested header, empty string if header name is not found
+    ///
+    std::string CURLGetHeader(const std::string &name)
+    {
+      if (!m_file)
+      {
+        kodi::Log(ADDON_LOG_ERROR, "kodi::vfs::CURLCreate(...) needed to call before!");
+        return false;
+      }
+      return ::kodi::addon::CAddonBase::m_interface->toKodi->kodi_filesystem->curl_get_header(::kodi::addon::CAddonBase::m_interface->toKodi->kodiBase, m_file, name);
```

Comment:
in case the c api returns NULL, you need to check for this here. IIRC, std::string has no ctor taking a nullptr